### DISCUSSION TITLE
How to apply routing when only a part of the UI should change?

### DISCUSSION THREAD

**Discussion by OrhanTozan** [vercel/next.js#37360] [Help] [answered]
I have the following Routing usecase:

/[userId].js UI should look like this:
```
<div>
    <Sidebar >
        <PostList />
    </Sidebar

    // Point X (Apply routing only here)

</div>
```

/[userId]/[postId].js UI should look like this:
```
<div>
    <Sidebar >
        <PostList />
    </Sidebar
    <Post id={postId}
</div>
```

It looks like I only want routing on Point X, not on top-level.

Having two seperate pages looks like overkill, because only part of the UI changes (just the selected post that is/is not visible)

How would I do this with NextJS?

**Comment by icyJoseph** [comment, marked as answer]
Either use a shared layout between two pages, `/[userid].js`and `/[userid]/[postid].js`, or do a catch all with:

`/[userId]/[[...postId]].js`  which will match `/foo/bar`, and also `/foo`, where `userId` is `foo` and `postId` is a list... which if not present, means you don't render that part of the UI, if its present, take the first element, ignore the rest, and render the `Point X`.

You might want to send/show a 404 page if there's more than one `postId`. Also the query object in the router, will be empty on first render, if not generated at build time.

I think that with this much info you'll figure it out.
